lib/wallets/wallet/impl/solana_wallet.dart

```diff
@@ -0,0 +1,362 @@
+
+import 'dart:math';
+
+import 'package:isar/isar.dart';
+import 'package:solana/dto.dart';
+import 'package:solana/solana.dart';
+import 'package:stackwallet/models/isar/models/blockchain_data/transaction.dart' as isar;
+import 'package:stackwallet/models/isar/models/isar_models.dart';
+import 'package:stackwallet/models/paymint/fee_object_model.dart';
+import 'package:stackwallet/utilities/amount/amount.dart';
+import 'package:stackwallet/utilities/default_nodes.dart';
+import 'package:stackwallet/utilities/enums/coin_enum.dart';
+import 'package:stackwallet/wallets/crypto_currency/coins/solana.dart';
+import 'package:stackwallet/wallets/crypto_currency/crypto_currency.dart';
+import 'package:stackwallet/wallets/models/tx_data.dart';
+import 'package:stackwallet/wallets/wallet/intermediate/bip39_wallet.dart';
+import 'package:stackwallet/utilities/logger.dart';
+import 'package:stackwallet/models/balance.dart';
+import 'package:tuple/tuple.dart';
+import 'package:stackwallet/utilities/enums/fee_rate_type_enum.dart';
+import 'package:stackwallet/models/node_model.dart';
+import 'package:stackwallet/services/node_service.dart';
+
+class SolanaWallet extends Bip39Wallet<Solana> {
+  SolanaWallet(CryptoCurrencyNetwork network) : super(Solana(network));
+
+  NodeModel? _solNode;
+
+  Future<Ed25519HDKeyPair> _getKeyPair() async {
+    return Ed25519HDKeyPair.fromMnemonic(await getMnemonic(), account: 0, change: 0);
+  }
+
+  Future<Address> _getCurrentAddress() async {
+    var addressStruct = Address(
+        walletId: walletId,
+        value: (await _getKeyPair()).address,
+        publicKey: List<int>.empty(),
+        derivationIndex: 0,
+        derivationPath: null,
+        type: cryptoCurrency.coin.primaryAddressType,
+        subType: AddressSubType.unknown);
+    return addressStruct;
+  }
+
+  Future<int> _getCurrentBalanceInLamports() async {
+    var rpcClient = RpcClient("${getCurrentNode().host}:${getCurrentNode().port}");
+    var balance = await rpcClient.getBalance((await _getKeyPair()).address);
+    return balance.value;
+  }
+
+  @override
+  FilterOperation? get changeAddressFilterOperation =>
+      throw UnimplementedError();
+
+  @override
+  Future<void> checkSaveInitialReceivingAddress() async {
+    try {
+      var address = (await _getKeyPair()).address;
+
+      await mainDB.updateOrPutAddresses([
+        Address(
+            walletId: walletId,
+            value: address,
+            publicKey: List<int>.empty(),
+            derivationIndex: 0,
+            derivationPath: null,
```

Comment:
Would this restore the same address as other popular Solana wallets?  Is there a standard derivation path or several which we should check?

Reply:
The ones in the _getKeyPair (So rest of the wallet) are fully compatible and same as popular Solana wallets like Phantom and Solflare.